File: src/main/java/net/swamphut/swampium/core/configs/ServiceSpecifyingConfig.java
package net.swamphut.swampium.core.configs;

import java.util.ArrayList;
import java.util.List;

public class ServiceSpecifyingConfig {
    public List<ServiceSpecifyingRule> specifyRules = new ArrayList<>();
    public List<ServiceSpecifyingRule> blacklistRules = new ArrayList<>();

    public class ServiceSpecifyingRule {
        public String requester = "";
        public List<ServiceProviderSpecifier> fullfillWith = new ArrayList<>();
    }

    public class ServiceProviderSpecifier {
        public String require = "";
        public String provider = "";
    }
}

File: src/main/kotlin/dev/reactant/reactant/core/configs/ServiceSpecifyingConfig.kt
package dev.reactant.reactant.core.configs

import java.util.*

data class ServiceSpecifyingConfig(
        var specifyRules: List<ServiceSpecifyingRule> = ArrayList(),
        var blacklistRules: List<ServiceSpecifyingRule> = ArrayList()
) {
    inner class ServiceSpecifyingRule {
        var requester = ""
        var fulfillWith: List<ServiceProviderSpecifier> = ArrayList()
    }

    inner class ServiceProviderSpecifier {
        var require = ""
        var provider = ""
    }
}
